src/Main.elm

```diff
@@ -67,19 +88,38 @@ init =
 -- Update
 
 
+toUpper : String -> String
+toUpper str =
+    String.toUpper (String.left 1 str) ++ String.dropLeft 1 str
+
+
+matchSelectedGenres : Model -> Movie -> Bool
+matchSelectedGenres model movie =
+    case Set.size model.selectedGenres of
+        0 ->
+            True
+
+        _ ->
+            Set.size (Set.intersect movie.genres model.selectedGenres) > 0
+
+
 type Msg
     = SelectMovie
     | MovieSelected ( Maybe Movie, List Movie )
+    | SelectGenre String
+    | RemoveGenre String
+    | MultiselectEvent Multiselect.Msg
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
 update msg model =
     case msg of
         SelectMovie ->
             ( model
-            , Random.generate
-                MovieSelected
-                (RandList.choose model.unwatched)
+            , model.unwatched
+                |> List.filter (matchSelectedGenres model)
+                |> RandList.choose
+                |> Random.generate MovieSelected
```

Comment:
@MattCheely I'm a fan of using the pipe/pizza pattern where possible. I.e., if there are two function calls in the same "line" then switch it to pizza. Not sure what the opinion/best practice is on this. Open to feedback.